src/attach-shadow.js

```diff
@@ -153,7 +137,8 @@ class ShadyRoot {
       for (let i=0, l=c$.length; i < l; i++) {
         const child = c$[i];
         const data = shadyDataForNode(child);
-        if (parentNode(child) === this.host && !data.assignedSlot) {
+        if (parentNode(child) === this.host &&
+            (child.localName === 'slot' || !data.assignedSlot)) {
```

Comment:
TODO: why is slot special handling required here?

Reply:
Elements are not composed into shadowRoot hosts. Therefore all non-rendering children must be manually removed at first render. This includes unassigned nodes and slots.